sagemaker/pipeline/pipeline_parameters.py

```diff
@@ -0,0 +1,764 @@
+        "--model-output-path",
+        type=str,
+        default="",
+        help="S3 path for model output. Default is determined by graph name and subpath.",
+    )
+    # TODO: Make this a pipeline param or derive from instance type during launch?
+    optional_args.add_argument(
+        "--num-trainers",
+        type=int,
+        default=4,
+        help="Number of trainers to use during training/inference. Set this to the number of GPUs."
+        "Default: 4",
+    )
+    required_args.add_argument(
+        "--train-inference-task",
+        type=str,
+        required=True,
+        help="Task type for training and inference, e.g. 'node_classification'. Required",
+    )
+    optional_args.add_argument(
+        "--train-yaml-s3",
+        type=str,
+        help="S3 path to train YAML configuration file. Required if you include train step.",
+    )
+    optional_args.add_argument(
+        "--use-graphbolt",
+        type=str,
+        default="false",
+        choices=["true", "false"],
+        help="Whether to use GraphBolt. Default: 'false'",
+    )
+
+    # Inference Configuration
+    optional_args.add_argument(
+        "--inference-yaml-s3",
+        type=str,
+        default=None,
+        help="S3 path to inference YAML configuration file. Default: same as --train-yaml-config",
+    )
+    # TODO: Need a good way to initialize this. Ideally "best-model" should pick up best epoch
+    # from training
+    optional_args.add_argument(
+        "--inference-model-snapshot",
+        type=str,
+        help="Which model snapshot to choose to run inference with, e.g. 'epoch-0'.",
+    )
+    optional_args.add_argument(
+        "--save-predictions",
+        action="store_true",
+        help="Whether to save predictions to S3 during inference.",
+    )
+    optional_args.add_argument(
+        "--save-embeddings",
+        action="store_true",
+        help="Whether to save embeddings to S3 during inference.",
+    )
+
+    # Script Paths
+    optional_args.add_argument(
+        "--dist-part-script",
+        type=str,
+        default=f"{SCRIPT_PATH}/../run/partition_entry.py",
+        help="Path to DistPartition SageMaker entry point script.",
+    )
+    optional_args.add_argument(
+        "--gb-convert-script",
+        type=str,
+        default=f"{SCRIPT_PATH}/../run/gb_convert_entry.py",
+        help="Path to GraphBolt partition script.",
+    )
+    optional_args.add_argument(
+        "--train-script",
+        type=str,
+        default=f"{SCRIPT_PATH}/../run/train_entry.py",
+        help="Path to training SageMaker entry point script.",
+    )
+    optional_args.add_argument(
+        "--inference-script",
+        type=str,
+        default=f"{SCRIPT_PATH}/../run/infer_entry.py",
+        help="Path to inference SageMaker entry point script.",
+    )
+    optional_args.add_argument(
+        "--gconstruct-script",
+        type=str,
+        default=f"{SCRIPT_PATH}/../run/gconstruct_entry.py",
```

Comment:
Where is the `../run/gconstruct_entry.py`?

Reply:
https://github.com/awslabs/graphstorm/blob/main/sagemaker/run/gconstruct_entry.py

Starting at `sagemaker/pipeline` which is where `SCRIPT_PATH` points to, we go up one level to `sagemaker/` then down to `sagemaker/run` where the entry points exist.

Reply:
OK